chef.py

```diff
@@ -0,0 +1,459 @@
+
+
+_NO_GITHUB_API_TOKEN_MESSAGE = """
+*******************************************************************************
+* Warning: GitHub API token not found in secrets.py. Scraping might be
+* rate-limited. See secrets.py.example to obtain a GitHub personal access token
+*******************************************************************************
+"""
+if not _GITHUB_API_TOKEN:
+    print(_NO_GITHUB_API_TOKEN_MESSAGE)
+
+sess = requests.Session()
+cache = FileCache('.webcache')
+forever_adapter = CacheControlAdapter(heuristic=CacheForeverHeuristic(), cache=cache)
+
+sess.mount('https://blockly-games.appspot.com', forever_adapter)
+sess.mount('http://fonts.googleapis.com', forever_adapter)
+sess.mount('https://raw.githubusercontent.com', forever_adapter)
+sess.mount('https://api.github.com', forever_adapter)
+
+
+headers = {
+    "User-Agent": "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:20.0) Gecko/20100101 Firefox/20.0",
+    "Accept-Encoding": "gzip, deflate",
+    "Connection": "keep-alive"
+}
+
+
+class BlocklyGamesChef(SushiChef):
+    """
+    The chef class that takes care of uploading channel to the content curation server.
+
+    We'll call its `main()` method from the command line script.
+    """
+    channel_info = {
+        'CHANNEL_SOURCE_DOMAIN': "blockly-games.appspot.com",
+        'CHANNEL_SOURCE_ID': "blockly-games",
+        'CHANNEL_TITLE': "Blockly Games",
+        'CHANNEL_THUMBNAIL': "thumbnail.png",
+        'CHANNEL_DESCRIPTION': "Teaches programming using block-based games, for children who have no prior experience with programming.",
+    }
+
+    def construct_channel(self, **kwargs):
+        """
+        Create ChannelNode and build topic tree.
+        """
+        # create channel
+        channel_info = self.channel_info
+        channel = nodes.ChannelNode(
+            source_domain = channel_info['CHANNEL_SOURCE_DOMAIN'],
+            source_id = channel_info['CHANNEL_SOURCE_ID'],
+            title = channel_info['CHANNEL_TITLE'],
+            thumbnail = channel_info.get('CHANNEL_THUMBNAIL'),
+            description = channel_info.get('CHANNEL_DESCRIPTION'),
+            language = "en",
+        )
+
+        download_all_languages(channel)
+        return channel
+
+
+def download_all_languages(channel):
+    """Download all available languages from Blockly."""
+    languages = []
+
+    with WebDriver("https://blockly-games.appspot.com", delay=1000) as driver:
+        for option in driver.find_elements_by_css_selector('#languageMenu option'):
+            blockly_language_code = option.get_attribute('value')
+            language_title = option.text
+            le_language_code = blockly_language_code
+
+            # There are some obscure languages that Blockly supports that we
+            # don't yet know about in le-utils. Skip those for now.
+            # TODO(davidhu): Add these language codes to le-utils
+            if le_language_code in ['hrx', 'pms', 'sco', 'be-tarask']:
```

Comment:
I'm working on a PR to add `sco` to `le-utils` soon, and I can add the others as well.

Reply:
Awesome!